roles/edpm_libvirt/tasks/configure.yml

```diff
@@ -94,3 +98,20 @@
   ansible.builtin.command: "restorecon -R /run/libvirt"
   changed_when: true
   when: run_libvirt.rc == 0
+
+- name: Copy TLS files to the compute node
+  tags:
+    - configure
+    - libvirt
+  become: true
+  loop:
+    - {"src": "{{ edpm_libvirt_certs_src }}/{{ inventory_hostname }}-tls.crt", "dest": "/etc/pki/libvirt/server-cert.pem"}
```

Comment:
don't the certs need to be under a ''libvirt'' subdirectory in case there are different certs for different services?

Reply:
so technially yes. nova and libvirt are just going to use the libvirt cert as we don't have any endpoint on the nova-compute agent but nova-compute need to talk to libvirt so it makes sense for nova and livbirt to share the same certs.

for ovn/ovs or other things like iscisd/multipathd they should likely have spereate certs so it would make sense to move these to a libvirt subdir in the src configmap.

Reply:
Just to clarify here, the destination obviously will be in /etc/pki/libvirt/... so thats in a libvirt directory.

The source is where the certs are mounted in the ansibleEE pod, and the ansibleEE pods are created per dataplane service.  That is, there is an ansibleEE pod for nova, and one for libvirt etc.  The certs needed in that pod are not differentiated per service (ie. they are under /var/lib/openstack/certs) because there are only ever certs for that one dataplane service

Does that make sense?

Reply:
It's correct, but I still think they should be mounted under a subdir with the service name, same as we do for the ConfigMaps and Secrets:

https://github.com/openstack-k8s-operators/dataplane-operator/blob/8d8a1051288a065e8fec16e07c907dc08ebbb4d8/pkg/deployment/deployment.go#L232

Reply:
introducing a dedicated role for certs managment sound like a good idea to me

Reply:
@slagle ok - I can change it accordingly.

@bogdando  @SeanMooney  The certs that nova and other services would need would be there in /var/lib/openstack/certs/{service} as long as those previous services have already executed.  Each service  copies the certs that are mounted to a specific directory which is then mounted (or not, in the case of libvirt) by the service container.  But this depends on an implied ordering (which probably already existed before TBH).  A dedicated dataplane service for cert management will make this all more predictable.

So, just to clarify, with the new dedicated dataplane service for cert management (lets call it "certs"), we will:
1. mount the secrets containing the certs/keys/cacerts for all nodes for all the enabled services that require certs
2. execute an ansible role that will copy the relevant certs/keys/cacerts  to /var/lib/openstack/certs/{service} or /var/lib/openstack/cacerts/{service} on each node. In the libvirt case, we'd move it to /etc/pki/libvirt etc.
3. each subsequent dataplane service can then use those certs as appropriate.  Most of them will simply mount the required certs in a container.   This simplifies the logic in each dataplane service to just doing a mount.

Is this what you guys had in mind?

Reply:
yep more or less
so for nova we would just mount it to the nova contaienrs
for libvirt we we just need to copy it to the correct location or update the libvirt config to know where it is.

so if just haveing the libvirt role copy/move the certs form

 /var/lib/openstack/cacerts/{libvirt} to /etc/pki/libvirt is all that is required we can have the libvirt role do that